paasta_tools/instance/kubernetes.py

```diff
@@ -499,6 +518,42 @@ def get_pod_status(pod: V1Pod) -> Dict[str, Any]:
     }
 
 
+def get_versions_for_pods(pods: Sequence[V1Pod],) -> List[KubernetesVersionDict]:
+    versions = []
+    pods_by_shas = get_pods_by_shas(pods)
+    for (git_sha, config_sha), pods in pods_by_shas.items():
+        versions.append(get_version_for_shas_and_pods(git_sha, config_sha, pods))
+    return versions
+
+
+def get_pods_by_shas(pods: Sequence[V1Pod]) -> Dict[Tuple[str, str], List[V1Pod]]:
+    pods_by_version: DefaultDict[Tuple[str, str], List[V1Pod]] = defaultdict(list)
+    for pod in pods:
+        git_sha = pod.metadata.labels["paasta.yelp.com/git_sha"]
+        config_sha = pod.metadata.labels["paasta.yelp.com/config_sha"]
+        pods_by_version[(git_sha, config_sha)].append(pod)
+    return pods_by_version
+
+
+def get_version_for_shas_and_pods(
+    git_sha: str, config_sha: str, pods: Sequence[V1Pod]
+) -> KubernetesVersionDict:
+    ready_pods = [pod for pod in pods if kubernetes_tools.is_pod_ready(pod)]
+    first_pod_created = min(
+        [pod.metadata.creation_timestamp.timestamp() for pod in pods]
+    )
+    return {
+        "name": None,
+        "type": "StatefulSet",
+        "replicas": len(pods),
+        "ready_replicas": len(ready_pods),
+        "create_timestamp": first_pod_created,
```

Comment:
A couple things I'd like feedback on here:
1.  it's a little misleading to give these version entries the type "StatefulSet" since they are actually subsets of the pods in the StatefulSet.  `"type": "StatefulSubset"`?  😛 
2. I'm using the min creation timestamp of the pods for a given version as the creation timestamp for the version...but that could easily be wrong with pod turnover.  But without this we can't really know which version is newer.

Reply:
1. Oh, I was just assuming type was just referencing what kind of k8s resource these pods belong to, not the type of this version 'object' itself, so it made perfect sense to stick as StatefulSet
2. Since our new status really only distinguishes the latest version as 'new' and the rest are all lumped up as 'old', it doesn't seem like it'd be too terrible a problem -- Do we know what the statefulset controller does during an update anyhow? If a older pod goes away before it is upgraded, does it get spun back up on the old version?

Reply:
You could probably use `ControllerRevision`s, which contain update metadata for statefulsets.

Reply:
Yep you are right, and we even get multiple ControllerRevisions during a bounce.  I changed it to use this.